archive/android_sparse.ksy

```diff
@@ -51,20 +51,25 @@ types:
          doc: CRC32 checksum of the original data
   image_header_entry:
     seq:
-      - id: chunk_type
-        type: u2
-        enum: chunk_types
-        doc: chunk type
-      - id: reserved
-        type: u2
-      - id: chunk_size
-        type: u4
-        doc: in blocks in output image
-      - id: total_size
-        type: u4
-        doc: in bytes of chunk input file including chunk header and data
+      - id: header
+        type: fixed_header
       - id: body
-        size: total_size - 12
+        size: header.total_size - header._sizeof
```

Comment:
@generalmimon, is there any benefit to use `sizeof<type>` in such cases? The fact that in this case we need `sizeof<type>` can probably be inferred automatically, when the type of the value is constant-size and can be resolved. Should we encourage usage of `_sizeof` in these cases in KSYs (but the compiler must generate the code as if `sizeof<type>` is used in these cases, maybe using `decltype`, where available, not lookups in runtime) because they give a bit more semantically-valuable information?